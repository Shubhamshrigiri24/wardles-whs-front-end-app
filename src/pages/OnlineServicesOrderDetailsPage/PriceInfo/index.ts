export { default, PriceInfo } from "./PriceInfo";
