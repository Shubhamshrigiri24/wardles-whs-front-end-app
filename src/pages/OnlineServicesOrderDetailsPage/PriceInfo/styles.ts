import { makeStyles, Theme } from "@material-ui/core";
import { wellColors } from "@welldigital/components";

export const useStyles = makeStyles((theme: Theme) => {
  return {
    divider: {
      backgroundColor: wellColors.elixir[900],
    },
    text: {
      color: wellColors.elixir[400],
      fontWeight: theme.typography.fontWeightRegular,
    },
    highlightedText: {
      fontSize: "18px",
      color: wellColors.elixir[900],
      fontWeight: theme.typography.fontWeightBold,
      letterSpacing: "0.15px",
    },
  };
});
