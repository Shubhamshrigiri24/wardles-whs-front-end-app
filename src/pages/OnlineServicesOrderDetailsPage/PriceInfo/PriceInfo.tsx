import React, { FC } from "react";
import { Grid, Divider } from "@material-ui/core";
import { Typography, Spacing } from "@welldigital/components";
import { useStyles } from "./styles";

export type PriceInfoProps = {
  label: string;
  price: string;
  highlightLabel?: boolean;
  highlightPrice?: boolean;
};
export const PriceInfo: FC<PriceInfoProps> = ({
  label,
  price,
  highlightLabel = false,
  highlightPrice = false,
}) => {
  const classes = useStyles();
  return (
    <>
      <Spacing spacing={3} />
      <Divider className={classes.divider} />
      <Spacing spacing={1} />
      <Grid container wrap={"nowrap"}>
        <Grid item xs>
          <Typography
            className={highlightLabel ? classes.highlightedText : classes.text}
            data-testid={"priceInfo/label"}
          >
            {label}
          </Typography>
        </Grid>
        <Grid item xs={"auto"}>
          <Typography
            className={highlightPrice ? classes.highlightedText : classes.text}
            data-testid={"priceInfo/price"}
          >
            {price}
          </Typography>
        </Grid>
      </Grid>
      <Spacing spacing={3} />
    </>
  );
};

export default PriceInfo;
