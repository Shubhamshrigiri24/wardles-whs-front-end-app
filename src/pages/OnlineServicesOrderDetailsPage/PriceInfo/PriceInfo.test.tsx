import React from "react";
import { render } from "@testing-library/react";
import PriceInfo, { PriceInfoProps } from "./PriceInfo";

const LABEL = "label";
const PRICE = "Â£5.99";

const makeProps = (props?: Partial<PriceInfoProps>): PriceInfoProps => {
  return {
    label: LABEL,
    price: PRICE,
    ...props,
  };
};

describe("<PriceInfo />", () => {
  it("should match snapshot", () => {
    const props = makeProps();
    const { container } = render(<PriceInfo {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    const props = makeProps();
    const { getByTestId } = render(<PriceInfo {...props} />);

    expect(getByTestId("priceInfo/label")).toHaveTextContent(LABEL);
    expect(getByTestId("priceInfo/price")).toHaveTextContent(PRICE);
  });

  it("should highlight label and price", () => {
    const props = makeProps({ highlightLabel: true, highlightPrice: true });
    const { getByTestId } = render(<PriceInfo {...props} />);

    expect(getByTestId("priceInfo/label").className).toContain(
      "highlightedText"
    );
    expect(getByTestId("priceInfo/price").className).toContain(
      "highlightedText"
    );
  });
});
