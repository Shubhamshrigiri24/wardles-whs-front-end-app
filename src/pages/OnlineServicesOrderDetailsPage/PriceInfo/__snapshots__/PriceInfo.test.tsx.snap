// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceInfo /> should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-spacing3-10"
    data-testid="spacing/root"
  />
  <hr
    class="MuiDivider-root makeStyles-divider-1"
  />
  <div
    class="makeStyles-spacing1-6"
    data-testid="spacing/root"
  />
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
    >
      <p
        class="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
        data-testid="priceInfo/label"
      >
        label
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto"
    >
      <p
        class="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
        data-testid="priceInfo/price"
      >
        Â£5.99
      </p>
    </div>
  </div>
  <div
    class="makeStyles-spacing3-10"
    data-testid="spacing/root"
  />
</div>
`;
